tests/disk/test_service.py

```diff
@@ -0,0 +1,203 @@
+        with pytest.raises(ExportException) as ex:
+            self.service.check_connected_devices()
+
+        assert ex.value.sdstatus is Status.LEGACY_USB_NOT_CONNECTED
+
+    def test_too_many_devices_connected(self):
+        self.mock_cli.get_connected_devices.return_value = [
+            SAMPLE_OUTPUT_USB,
+            "/dev/sdb",
+        ]
+        with pytest.raises(ExportException) as ex:
+            self.service.check_connected_devices()
+
+        assert ex.value.sdstatus is Status.LEGACY_USB_ENCRYPTION_NOT_SUPPORTED
+
+    def test_device_is_not_luks(self):
+        self.mock_cli.is_luks_volume.return_value = False
+
+        # When VeraCrypt is supported, this will no longer be an exception
+        # and the return status will change
+        with pytest.raises(ExportException) as ex:
+            self.service.check_disk_format()
+
+        assert ex.value.sdstatus is Status.LEGACY_USB_ENCRYPTION_NOT_SUPPORTED
+
+    def test_check_usb_error(self):
+        self.mock_cli.get_connected_devices.side_effect = ExportException(
+            sdstatus=Status.LEGACY_ERROR_USB_CHECK
+        )
+
+        with pytest.raises(ExportException) as ex:
+            self.service.check_connected_devices()
+
+        assert ex.value.sdstatus is Status.LEGACY_ERROR_USB_CHECK
+
+    def test_check_disk_format(self):
+        status = self.service.check_disk_format()
+
+        assert status is Status.LEGACY_USB_ENCRYPTED
+
+    def test_check_disk_format_error(self):
+        self.mock_cli.get_partitioned_device.side_effect = ExportException(
+            sdstatus=NewStatus.INVALID_DEVICE_DETECTED
+        )
+
+        with pytest.raises(ExportException) as ex:
+            self.service.check_disk_format()
+
+        # We still return the legacy status for now
+        assert ex.value.sdstatus is Status.LEGACY_USB_ENCRYPTION_NOT_SUPPORTED
+
+    def test_export(self):
+        # Currently, a successful export does not return a success status.
+        # When the client is updated, this will change to assert EXPORT_SUCCESS
+        # is returned.
+        self.service.export()
+
+    def test_export_disk_not_supported(self):
+        self.mock_cli.is_luks_volume.return_value = False
+
+        with pytest.raises(ExportException) as ex:
+            self.service.export()
+
+        assert ex.value.sdstatus is Status.LEGACY_USB_ENCRYPTION_NOT_SUPPORTED
+
+    def test_export_write_error(self):
+        self.mock_cli.is_luks_volume.return_value = True
+        self.mock_cli.write_data_to_device.side_effect = ExportException(
+            sdstatus=Status.LEGACY_ERROR_USB_WRITE
+        )
+
+        with pytest.raises(ExportException) as ex:
+            self.service.export()
+
+        assert ex.value.sdstatus is Status.LEGACY_ERROR_USB_WRITE
+
+    def test_export_throws_new_exception_return_legacy_status(self):
+        self.mock_cli.get_connected_devices.side_effect = ExportException(
+            sdstatus=NewStatus.ERROR_MOUNT
+        )
```

Comment:
Naming nit: In general terms, I think that naming all mocks `mock_*` is unnecessary. The readability improvement is debatable, and the `mock_*` prefix doesn't add any new information. From the moment you assign to it or assert on invocations, the thing is being treated as controlled entity, a mock, anyway.

I would say that this block is meant to be understood as: 

"Given `self.cli.get_connected_devices`'s side effect is `ExportException(...)`"

And that 1. is closer to that than 2.:

```
1. self.cli.get_connected_devices.side_effect = ExportException(...) 

2. self.mock_cli.get_connected_devices.side_effect = ExportException(...) 
```

Isn't it?

Reply:
I see value in indicating which components are stubs/mocks and which ones are 'live', personally, just to make it explicit that we are using a 'fake' cli and not the real one.

Reply:
Fair. That's something we can talk about if we'd prefer having a consistent convention. I'd usually not prefix variables, and go for shorter names the closer usage is from the declaration. (In Python I don't go all the way to the one letter convention in Go, but in my experience that scope-size-to-name-length heuristic does improve readability, so I follow that spirit.)